src/main/java/com/shopify/model/ShopifyProductUpdateRequest.java

```diff
@@ -148,57 +148,55 @@ public PublishedStep withVariantRequests(final List<ShopifyVariantRequest> varia
 				changed = true;
 			}
 
-			final List<ShopifyVariant> shopifyVariants = new ArrayList<>(variantRequests.size());
-			final List<Integer> positions = new ArrayList<>(variantRequests.size());
+			List<ShopifyVariant> shopifyVariants = new ArrayList<>(variantRequests.size());
+			List<Integer> positions = new ArrayList<>(variantRequests.size());
 
 			for (int i = 0; i < variantRequests.size(); i++) {
 				final ShopifyVariantRequest shopifyVariantRequestForPosition = variantRequests.get(i);
 				positions.add(shopifyVariantRequestForPosition.getRequest().getPosition());
-
 			}
 
-			int maxPosition = variantRequests.stream().map(ShopifyVariantRequest::getRequest)
-					.map(ShopifyVariant::getPosition).max(Comparator.naturalOrder()).get();
+			int maxPosition = positions.stream().max(Comparator.naturalOrder()).get();
+
 			Collections.sort(variantRequests, new ShopifyVariantRequestOption1Comparator());
+
+			updateVariants(variantRequests, shopifyVariants, positions, maxPosition);
+
+			shopifyProduct.setVariants(shopifyVariants);
+
+			return this;
+		}
+
+		private void updateVariants(final List<ShopifyVariantRequest> variantRequests,
```

Comment:
Can you rename this function to be something like mapVariantRequestToShopifyVariant or something like that? For a second here i though we were updating variants right here but it's just mapping them within the context of a product.